src/commonTest/kotlin/com/charleskorn/kaml/YamlReadingDuplicatesTest.kt

```diff
@@ -0,0 +1,198 @@
+package com.charleskorn.kaml
+
+import ch.tutteli.atrium.api.fluent.en_GB.message
+import ch.tutteli.atrium.api.fluent.en_GB.notToThrow
+import ch.tutteli.atrium.api.fluent.en_GB.toBe
+import ch.tutteli.atrium.api.fluent.en_GB.toThrow
+import ch.tutteli.atrium.api.verbs.expect
+import kotlinx.serialization.decodeFromString
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class YamlReadingDuplicatesTest : Spek({
+    describe("reading YAML objects") {
+        val parser = Yaml.default
+
+        context("without duplicates") {
+            val yaml = """
+                key1: {}
+                key2: {}
+            """.trimIndent()
+
+            it("does not throw exceptions") {
+                expect({ parser.decodeFromString<MapStructure>(yaml) }).notToThrow()
+            }
+        }
+
+        context("with a single duplicate of a single key") {
+            val yaml = """
+                key1: {}
+                key1: {}
+            """.trimIndent()
+
+            it("throws an exception with a correct message") {
+                expect({ parser.decodeFromString<MapStructure>(yaml) }).toThrow<DuplicateKeysException> {
+                    message.toBe(
+                        """
+                            Found duplicates of key 'key1' defined at line 1, column 1:
```

Comment:
Perhaps the error message would be clearer if it read as something like this:

```
Key 'key1' is only allowed once, but is defined multiple times:
- at line 1, column 1
- at line 2, column 1
```

What do you think?

Reply:
Yes, this looks better. I didn't have a good idea for the message.